src/Components/UI/VerticalNavigation/VerticalNavigation.jsx

```diff
@@ -0,0 +1,72 @@
+import { NavLink } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { useFetch } from "../../../CustomHooks";
+import "./VerticalNavigation.css";
+const VerticalNavigation = ({ verticalNavOpen }) => {
+  const navClasses = verticalNavOpen ? " vertical-navigation--active" : "";
+
+  const { data, serverCall: fetchCategories } = useFetch();
+
+  useEffect(() => {
+    fetchCategories({ method: "get", url: "/api/categories" });
+  }, []);
+
+  const categoriesEl = data.categories
+    ? data.categories.map((category) => {
+        return (
+          <li className="menu-item" key={category._id}>
+            <i className="fa-solid fa-play"></i>
+            <span>{category.categoryName}</span>
+          </li>
+        );
+      })
+    : null;
+
+  return (
+    <aside className={`vertical-navigation ${navClasses}`}>
+      <ul className="menu no-list-style">
+        <li className="menu-item">
+          <i className="fa-solid fa-house"></i>
+          <NavLink to="/">
+            Home
+          </NavLink>
+        </li>
```

Comment:
These element are repeating just the icon and text are changing make it into a component